portal_recaptcha/models/res_config.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2019 Simone Orsi - Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+
+
+class PortalConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    recaptcha_key_site = fields.Char()
+    recaptcha_key_secret = fields.Char()
+
+    @api.model
+    def get_values(self):
+        res = super(PortalConfigSettings, self).get_values()
+        res.update(
+            recaptcha_key_site=self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("portal_recaptcha.recaptcha_key_site"),
+            recaptcha_key_secret=self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("portal_recaptcha.recaptcha_key_secret"),
+        )
+        return res
+
+    @api.multi
+    def set_values(self):
+        super(PortalConfigSettings, self).set_values()
+        param = self.env["ir.config_parameter"].sudo()
+        param.set_param("portal_recaptcha.recaptcha_key_site", self.recaptcha_key_site)
+        param.set_param(
+            "portal_recaptcha.recaptcha_key_secret", self.recaptcha_key_secret
+        )
```

Comment:
I have not written the doc yet but from official doc : 

```py
class ResConfigSettings(models.TransientModel, ResConfigModuleInstallationMixin):
    """ Base configuration wizard for application settings.  It provides support for setting
        default values, assigning groups to employee users, and installing modules.
        To make such a 'settings' wizard, define a model like::

            class MyConfigWizard(models.TransientModel):
                _name = 'my.settings'
                _inherit = 'res.config.settings'

                default_foo = fields.type(..., default_model='my.model'),
                group_bar = fields.Boolean(..., group='base.group_user', implied_group='my.group'),
                module_baz = fields.Boolean(...),
                config_qux = fields.Char(..., config_parameter='my.parameter')
                other_field = fields.type(...),

        The method ``execute`` provides some support based on a naming convention:

        *   For a field like 'default_XXX', ``execute`` sets the (global) default value of
            the field 'XXX' in the model named by ``default_model`` to the field's value.
...
        *   For a field with no specific prefix BUT an attribute 'config_parameter',
            ``execute`` will save its value in an ir.config.parameter (global setting for the
            database).
...
    """
```

Donc dans ton cas, deux options : 

1. `config_qux = fields.Char(..., config_parameter='my.parameter')`
2. other_field = fields.type(...),

La première solution a le même résultat que la tienne mais Odoo crée les fonctions `set_values` et `get_values` lui-même.

La deuxième solution permet de faire ce qu'on veut des valeurs du formulaire, pour les écrire sur la company par exemple. Mais je ne pense pas que ce soit nécessaire.

Dans ton cas, tu pourrais simplifier en : 


```py
class PortalConfigSettings(models.TransientModel):
    _inherit = "res.config.settings"

    recaptcha_key_site = fields.Char(config_parameter ='portal_recaptcha.recaptcha_key_site')
    recaptcha_key_secret = fields.Char(config_parameter ='portal_recaptcha.recaptcha_key_secret')
```

Reply:
Fixed in the PR mentionned below